### PR TITLE
support creation of CQ and persistence of corresponding meta

### LINKED ISSUE #282: support creation, stop and query of continuous query
### Describe your question

# Create

## Create continues query through ts-sql

```sql
CREATE CONTINUOUS QUERY <cq_name> ON <database_name>
RESAMPLE EVERY <interval> FOR <interval>
BEGIN
  <cq_query>
END
```

- The number of CQs must be limited. Number of DataNode x 100.
- `<cq_name>` should be globally unique.
- Check whether the target `<db>`,`<rp>` exists.
- Check whether the target `measurement` exists and whether the `measurement`conflicts with the current CQ.
- The output of a continuous query must go to either a different measurement or to a different retention policy.
- In theory they'd still be able to create a cycle with multiple continuous queries. We should check for these and disallow.
- The CQ execution frequency cannot be less than `1s`.

## ts-meta persistence continues query task

[openGemini/open_src/influx/meta/database.go](https://github.com/openGemini/openGemini/blob/8674137a4b9a9cd43475dfaf324e5e84013e7525/open_src/influx/meta/database.go#L18-L24)

```go
type DatabaseInfo struct {
	......
  ContinuousQueries []ContinuousQueryInfo # ⭐️
}
```

```go
type ContinuousQueryInfo struct {
  Name string
  Query string
    ......
}
```

- The `ContinuousQueries` field is added, please ensure upgrade compatibility.

## additional

- every time CQ completes a cycle, it needs to check whether the task is deleted. You also need to check whether the load can be balanced to other SQL nodes.

# Query task

```sql
> SHOW CONTINUOUS QUERIES

name: _internal
name    query
----    -----
cq_demo  CREATE CONTINUOUS QUERY ......
```

Get CQ tasks from Meta.

# Stop

```sql
DROP CONTINUOUS QUERY <cq_name> ON <database_name>
```

1. Delete the CQ task from Meta. Save CQ records to tombstone.
2. SQL detects that CQ has been deleted, so it will not be executed in the next cycle.

### PR DESCRIPTION
**Pull request by Benevor**
### What problem does this PR solve?

Issue Number: resolve #282

### What is changed and how it works?
Implement the construction, sending, and application of CreateContinuousQueryCommand.

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B
- [x] Test cases to be added
- [ ] No code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open_src/influx/meta/proto/meta.pb.go [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-// source: open_src/influx/meta/proto/meta.proto
+// source: meta.proto
 
 package proto
 
 import (
 	fmt "fmt"
-	math "math"
-
 	proto "github.com/gogo/protobuf/proto"
```
Comment (xmh1011) [inline open_src/influx/meta/proto/meta.pb.go:8]: fmt import

Comment (Benevor) [reply open_src/influx/meta/proto/meta.pb.go]: I have changed the import order.

File: open_src/influx/coordinator/statement_executor.go [hunk 2]
```diff
@@ -145,6 +145,11 @@ func (e *StatementExecutor) ExecuteStatement(stmt influxql.Statement, ctx *query
 			messages = append(messages, query.ReadOnlyWarning(stmt.String()))
 		}
 		err = e.executeCreateRetentionPolicyStatement(stmt)
+	case *influxql.CreateContinuousQueryStatement:
+		if ctx.ReadOnly { // Benevor TODO: for what?
```
Comment (shilinlee) [inline open_src/influx/coordinator/statement_executor.go:149]: If it is ReadOnly, it usually means that there is not enough disk space.

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: open_src/influx/coordinator/statement_executor.go (hunk 2 above)
Comment (Benevor) [reply open_src/influx/coordinator/statement_executor.go:149]: I understand.

File: open_src/influx/meta/continuous_query.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	// Mark whether this cq has been deleted
+	MarkDeleted bool
+}
+
+// NewContinuousQueryInfo returns a new instance of ContinuousQueryInfo
+// with default values.
+func NewContinuousQueryInfo() *ContinuousQueryInfo {
+	return &ContinuousQueryInfo{
+		Name:        "",
```
[the remark this answers is not in the document]
Comment (Benevor) [reply open_src/influx/meta/continuous_query.go:24]: Yest, it is.

File: open_src/influx/meta/continuous_query.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		Name:        proto.String(cqi.Name),
+		Query:       proto.String(cqi.Query),
+		MarkDeleted: proto.Bool(cqi.MarkDeleted),
+	}
+
+	return pb
+}
+
+// unmarshal deserializes from a protobuf representation.
+func (cqi *ContinuousQueryInfo) unmarshal(pb *proto2.ContinuousQueryInfo) {
```
[the remark this answers is not in the document]
Comment (Benevor) [reply open_src/influx/meta/continuous_query.go:61]: `unmarshal()` will only be used in this package, but `Marshal()` will be called by other packages.

**Review by scuzyj — COMMENTED** [review]
It is recommended that the show continues query command be implemented as soon as possible so that test cases can be written for overwriting verification.

**Pull request merged**